t/sys/chdir.c

```diff
@@ -0,0 +1,361 @@
+    ok(str != NULL && strcmp(str, unifyfs_root) == 0,
+       "%s:%d getcwd returned %s expected %s",
+       __FILE__, __LINE__, str, unifyfs_root);
+#endif
+
+    /* change to root directory */
+    rc = chdir("/");
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, "/", strerror(errno));
+
+    /* check that we're in root directory */
+    str = get_current_dir_name();
+    ok(str != NULL, "%s:%d get_current_dir_name: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, "/") == 0,
+       "%s:%d get_current_dir_name returned %s expected %s",
+       __FILE__, __LINE__, str, "/");
+    if (str != NULL) {
+        free(str);
+    }
+
+    /* change to unifyfs root directory */
+    rc = chdir(unifyfs_root);
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, unifyfs_root, strerror(errno));
+
+    /* check that we're in unifyfs root directory */
+    str = get_current_dir_name();
+    ok(str != NULL, "%s:%d get_current_dir_name: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, unifyfs_root) == 0,
+       "%s:%d get_current_dir_name returned %s expected %s",
+       __FILE__, __LINE__, str, unifyfs_root);
+    if (str != NULL) {
+        free(str);
+    }
+
+    /* change to unifyfs/dir1 */
+    rc = chdir(buf);
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, buf, strerror(errno));
+
+    /* check that we're in unifyfs/dir1 */
+    str = get_current_dir_name();
+    ok(str != NULL, "%s:%d get_current_dir_name: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, buf) == 0,
+       "%s:%d get_current_dir_name returned %s expected %s",
+       __FILE__, __LINE__, str, buf);
+    if (str != NULL) {
+        free(str);
+    }
+
+    /* change back to root unifyfs directory */
+    rc = chdir("..");
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, "..", strerror(errno));
+
+    /* check that we're in root unifyfs directory */
+    str = get_current_dir_name();
+    ok(str != NULL, "%s:%d get_current_dir_name: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, unifyfs_root) == 0,
+       "%s:%d get_current_dir_name returned %s expected %s",
+       __FILE__, __LINE__, str, unifyfs_root);
+    if (str != NULL) {
+        free(str);
+    }
+
+    /* change to unifyfs/dir2 directory */
+    rc = chdir("dir2");
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, "dir2", strerror(errno));
+
+    /* check that we're in unifyfs/dir2 */
+    str = get_current_dir_name();
+    ok(str != NULL, "%s:%d get_current_dir_name: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, buf2) == 0,
+       "%s:%d get_current_dir_name returned %s expected %s",
+       __FILE__, __LINE__, str, buf2);
+    if (str != NULL) {
+        free(str);
+    }
+
+
+#if 0
+    /* change to root directory */
+    rc = chdir("/");
```

Comment:
This one as well?

Reply:
I'd like to activate these tests once our directory wrappers are working.  Added a TODO.

Reply:
Sounds good.